app/src/main/java/com/example/bookapisearch/view/BookDisplayFragment.kt

```diff
@@ -54,30 +60,67 @@ class BookDisplayFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         displayBinding.rvBookResult.layoutManager = GridLayoutManager(context, 2)
         displayBinding.rvBookResult.adapter = adapter
+        setupObservables()
     }
 
-    override fun onStart() {
-        super.onStart()
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
-            arguments?.getParcelable(BOOK_SEARCH_ARGS, BookSearchArguments::class.java)?.let {
-                repository.searchBookByBookTitle(it.bookTitle, it.bookType, it.maxResults) {
-                    // todo update adapter
-                    adapter.updateDataSet(
-                        it.items?.map { bookItems ->
+    private val observerNetworkConnectivity: Observer<BookResponse> = Observer {
+        Log.d("TONY", "Observing Forever: ")
+    }
+
+    /**
+     *
+     */
+    private fun setupObservables() {
+        bookViewModel.bookDataResult.observeForever(observerNetworkConnectivity)
+
+        bookViewModel.bookDataResult.observe(viewLifecycleOwner) {
+            adapter.updateDataSet(
+                it.items?.map { bookItems ->
                             BookVolumeInfo(
                                 bookItems.volumeInfo?.title,
                                 bookItems.volumeInfo?.imageLinks
                             )
                         } ?: emptyList()
-                    )
-                }
+            )
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        bookViewModel.bookDataResult.removeObserver(observerNetworkConnectivity)
+    }
 
+    override fun onStart() {
+        super.onStart()
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            arguments?.getParcelable(BOOK_SEARCH_ARGS, BookSearchArguments::class.java)?.let {
+                bookViewModel.searchBooksByArgs(
+                    it.bookTitle,
+                    it.bookType,
+                    it.maxResults
+                )
+//                repository.searchBookByBookTitle(it.bookTitle, it.bookType, it.maxResults) {
```

Comment:
You should not included commented code in a Pull Request. @AntoninoAN  please fix it!.

Reply:
@AntoninoAN  already fixed in the new commit, Thank you for the observation.